File: data/EXP01/golang_no_system_var/2502/2502.output_3.go
type Allocator struct {
    memory []int
}

func Constructor(n int) Allocator {
    memory := make([]int, n)
    return Allocator{memory: memory}
}

func (a *Allocator) Allocate(size int, mID int) int {
    for i := 0; i <= len(a.memory)-size; i++ {
        if a.memory[i] == 0 {
            isBlockAvailable := true
            for j := i; j < i+size; j++ {
                if a.memory[j] != 0 {
                    isBlockAvailable = false
                    break
                }
            }
            if isBlockAvailable {
                for j := i; j < i+size; j++ {
                    a.memory[j] = mID
                }
                return i
            }
        }
    }
    return -1
}

func (a *Allocator) Free(mID int) int {
    count := 0
    for i := 0; i < len(a.memory); i++ {
        if a.memory[i] == mID {
            a.memory[i] = 0
            count++
        }
    }
    return count
}

File: data/EXP01/python_3/2502/2502.output_3.py
# @lc app=leetcode id=2502 lang=python
class Allocator:

    def __init__(self, n: int):
        self.memory = [None] * n
        self.allocations = {}

    def allocate(self, size: int, mID: int) -> int:
        for i in range(len(self.memory)):
            if self.memory[i] is None:
                if i + size <= len(self.memory):
                    if all(x is None for x in self.memory[i:i+size]):
                        for j in range(i, i+size):
                            self.memory[j] = mID
                        if mID not in self.allocations:
                            self.allocations[mID] = []
                        self.allocations[mID].append((i, size))
                        return i
        return -1

    def free(self, mID: int) -> int:
        if mID in self.allocations:
            freed_memory = 0
            for allocation in self.allocations[mID]:
                start, size = allocation
                for i in range(start, start+size):
                    if self.memory[i] == mID:
                        self.memory[i] = None
                        freed_memory += 1
            del self.allocations[mID]
            return freed_memory
        else:
            return 0